record/record.go

```diff
@@ -144,23 +146,29 @@ func (r *Reader) nextChunk(wantFirst bool) error {
 			chunkType := r.buf[r.j+6]
 
 			if checksum == 0 && length == 0 && chunkType == 0 {
-				if wantFirst {
+				if wantFirst || r.recovering {
 					// Skip the rest of the block, if it looks like it is all zeroes.
 					// This is common if the record file was created via mmap.
-					r.i = r.n
-					r.j = r.n
+					r.Recover()
```

Comment:
If you get here the original way, when wantFirst and !r.recovering, then r.err should be nil, and then r.Recover is a no-op, so you'll infinite loop because r.i and r.j don't move, right?

If so, you should probably write a test for that. :-)